pkg/parsers/operatingsystem/operatingsystem_windows.go

```diff
@@ -1,44 +1,40 @@
 package operatingsystem // import "github.com/docker/docker/pkg/parsers/operatingsystem"
 
 import (
-	"unsafe"
+	"fmt"
 
-	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
 )
 
-// See https://code.google.com/p/go/source/browse/src/pkg/mime/type_windows.go?r=d14520ac25bf6940785aabb71f5be453a286f58c
-// for a similar sample
-
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
 
-	var h windows.Handle
-
 	// Default return value
 	ret := "Unknown Operating System"
 
-	if err := windows.RegOpenKeyEx(windows.HKEY_LOCAL_MACHINE,
-		windows.StringToUTF16Ptr(`SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\`),
-		0,
-		windows.KEY_READ,
-		&h); err != nil {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\WIndows NT\CurrentVersion`, registry.QUERY_VALUE)
```

Comment:
Should `pkg/parsers/kernel/` also be updated to use the `registry` package? https://github.com/moby/moby/blob/4f0d95fa6ee7f865597c03b9e63702cdcb0f7067/pkg/parsers/kernel/kernel_windows.go#L33-L53

Reply:
Sure but orthogonal to this PR itself. Can be addressed as a follow-up

Reply:
oh, definitely not needed in this PR :+1: